src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,239 @@
+using Aquifer.Common.Tiptap;
+using Aquifer.Common.Utilities;
+using Aquifer.Data;
+using Aquifer.Data.Entities;
+using Azure.Storage.Queues.Models;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Aquifer.Jobs.Subscribers;
+
+public sealed class ResourceContentVersionSimilarityMessageSubscriber(
+    AquiferDbContext _dbContext,
+    ILogger<ResourceContentVersionSimilarityMessageSubscriber> _logger)
+{
+    [Function(nameof(ScoreResourceContentVersionSimilarityMessageSubscriber))]
+    public async Task ScoreResourceContentVersionSimilarityMessageSubscriber(
+        [QueueTrigger(Queues.GenerateResourceContentSimilarityScore)]
+        QueueMessage queueMessage,
+        CancellationToken ct)
+    {
+        await queueMessage.ProcessAsync<ScoreResourceContentVersionSimilarityMessage, ResourceContentVersionSimilarityMessageSubscriber>(
+            _logger,
+            nameof(ScoreResourceContentVersionSimilarityMessageSubscriber),
+            ProcessAsync,
+            ct);
+    }
+
+    private async Task ProcessAsync(QueueMessage queueMessage, ScoreResourceContentVersionSimilarityMessage message, CancellationToken ct)
+    {
+        var similarityScoreEntity = await GenerateResourceContentVersionSimilarityScoreEntity(message, _dbContext, ct);
+        
+        await _dbContext.ResourceContentVersionSimilarityScores.AddAsync(similarityScoreEntity, ct);
+        await _dbContext.SaveChangesAsync(ct);
+    }
+    
+    private async Task<ResourceContentVersionSimilarityScoreEntity> GenerateResourceContentVersionSimilarityScoreEntity(
+        ScoreResourceContentVersionSimilarityMessage message,
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        ResourceContentVersionMachineTranslationEntity machineTranslationEntity;
+        var similarityScoreEntity = new ResourceContentVersionSimilarityScoreEntity { BaseVersionId = message.BaseVersionId };
+        string baseVersionText;
+        string compareVersionText;
+        
+        switch (message.ComparisonType)
+        {
+            case ResourceContentVersionSimilarityComparisonType.MachineTranslationToResourceContentVersion:
+                similarityScoreEntity.BaseVersionType = ResourceContentVersionTypes.Base;
+                similarityScoreEntity.ComparedVersionType = ResourceContentVersionTypes.MachineTranslation;
+
+                if (message.CompareVersionId is null)
+                {
+                    machineTranslationEntity = await GetMachineTranslationForResourceContentVersion(
+                        message.BaseVersionId,
+                        dbContext,
+                        ct);
+                    
+                    similarityScoreEntity.ComparedVersionId = machineTranslationEntity.Id;
+                }
+                else
+                {
+                    similarityScoreEntity.ComparedVersionId = (int)message.CompareVersionId;
```

Comment:
Don't cast a nullable to non-nullable.  For `Nullable<T>` value types you can use the `Value` property instead.
```suggestion
                    similarityScoreEntity.ComparedVersionId = message.CompareVersionId.Value;
```

Reply:
(same below)